HeterogeneousCore/AlpakaInterface/interface/OneToManyAssoc.h

```diff
@@ -0,0 +1,276 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_OneToManyAssoc_h
+#define HeterogeneousCore_AlpakaInterface_interface_OneToManyAssoc_h
+
+#include <alpaka/alpaka.hpp>
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <type_traits>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/prefixScan.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/AtomicPairCounter.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/FlexiStorage.h"
+
+namespace cms {
+  namespace alpakatools {
+
+    template <typename I,    // type stored in the container (usually an index in a vector of the input values)
+              int32_t ONES,  // number of "Ones"  +1. If -1 is initialized at runtime using external storage
+              int32_t SIZE   // max number of element. If -1 is initialized at runtime using external storage
+              >
+    class OneToManyAssocBase {
+    public:
+      using Counter = uint32_t;
+
+      using CountersOnly = OneToManyAssocBase<I, ONES, 0>;
+
+      using index_type = I;
+
+      struct View {
```

Comment:
I understand this is a 1:1 translation of the CUDA code, so this clean up couldmalso come later.
Anyway
  - I would prefer to rename this from `View` to `Args` or something that is less confusing with the "View" we have in the SoA and the "View" we have in alpaka :-)
  - in fact, it might be less confusing to just pass the 5 parameters independently, instead of wrapping them in a `struct` ?